### PR TITLE
[Translation] skip tests if required class does not exist

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
Builds on AppVeyor only fail for `master` for some reason that I do not understand but this logic applies to `3.4` too.

**Comment by xabbuh** [comment]
ah, looks like the failure was actually caused by #37371 which was merged into `master`

File: src/Symfony/Component/Translation/Tests/TranslatorCacheTest.php [hunk 1]
```diff
@@ -59,6 +59,10 @@ protected function deleteTmpDir()
      */
     public function testThatACacheIsUsed($debug)
     {
+        if (!class_exists(\MessageFormatter::class)) {
+            $this->markTestSkipped(sprintf('Skipping test as the "%s" class does not exist.', \MessageFormatter::class));
```
Comment (derrabus) [inline src/Symfony/Component/Translation/Tests/TranslatorCacheTest.php:63]: The class does not exist when `ext-intl` is missing. I think the message should reflect that.

Comment (xabbuh) [reply src/Symfony/Component/Translation/Tests/TranslatorCacheTest.php]: Good idea, I have updated the message to provide more information.

Comment (xabbuh) [reply src/Symfony/Component/Translation/Tests/TranslatorCacheTest.php]: Suggesting the PHP extension or the polyfill package is inline with the exception that is thrown by the `IntlFormatter`.

**Comment by xabbuh** [comment]
everything's back to green \o/

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**